src/flask_pyoidc/provider_configuration.py

```diff
@@ -164,22 +181,45 @@ def ensure_provider_metadata(self):
     def registered_client_metadata(self):
         return self._client_metadata
 
-    def register_client(self, redirect_uris, extra_parameters=None):
+    def register_client(self, client: Client, redirect_uri: str):
+
         if not self._client_metadata:
-            if 'registration_endpoint' not in self._provider_metadata:
+            if not self._provider_metadata['registration_endpoint']:
                 raise ValueError("Can't use dynamic client registration, provider metadata is missing "
                                  "'registration_endpoint'.")
 
             registration_request = self._client_registration_info.to_dict()
-            registration_request['redirect_uris'] = redirect_uris
-            if extra_parameters:
-                registration_request.update(extra_parameters)
-
-            resp = self.requests_session \
-                .post(self._provider_metadata['registration_endpoint'],
-                      json=registration_request,
-                      timeout=self.DEFAULT_REQUEST_TIMEOUT)
-            self._client_metadata = ClientMetadata(redirect_uris=redirect_uris, **resp.json())
+            # Check if the user has passed list of redirect_uris in ClientRegistrationInfo.
+            if 'redirect_uris' not in registration_request:
+                # If not, create it.
+                registration_request['redirect_uris'] = [redirect_uri]
+            # if it's passed, check if it's type list.
+            elif not isinstance(registration_request['redirect_uris'], list):
+                raise ValueError('redirect_uris must be a list, not str.')
+            else:
+                # Add redirect_uri into the list of redirect_uris passed by
+                # the user in ClientRegistrationInfo.
+                registration_request['redirect_uris'].append(redirect_uri)
+            post_logout_redirect_uris = []
+            # Check if post_logout_redirect_uris exists.
+            if 'post_logout_redirect_uris' in registration_request:
+                post_logout_redirect_uris = registration_request['post_logout_redirect_uris']
+                # If it exists, add this into the list of redirect_uris.
+                registration_request['redirect_uris'].extend(post_logout_redirect_uris)
+            # Remove duplicate redirect_uris from the list.
+            registration_request['redirect_uris'] = list(set(registration_request['redirect_uris']))
+
+            # Send request to register the client dynamically.
+            registration_response = client.register(
+                url=self._provider_metadata['registration_endpoint'],
+                **registration_request)
+            logger.debug(registration_response.to_dict())
+            # post_logout_redirect_uris is not returned to the request so
```

Comment:
This also seems like something provider specific (e.g. Keycloak), I don't think it should be needed for a provider properly supporting `post_logout_redirect_uris`.

Reply:
`post_logout_redirect_uris` parameter is not a part of openid specs so no IdP supports it. To make it easy for the user, `ClientMetada` and pyoidc support `post_logout_redirect_uris` as an argument.

Reply:
Same as above, `post_logout_redirect_uris` is added to Client Registration data by a separate spec.
So they shouldn't be mixed.